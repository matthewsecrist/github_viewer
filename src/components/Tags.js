import React from 'react'

import { Tag } from 'rbx'

const chooseColor = color => {
  var r = parseInt(color.substr(0, 2), 16)
  var g = parseInt(color.substr(2, 2), 16)
  var b = parseInt(color.substr(4, 2), 16)
  var c = (r * 299 + g * 587 + b * 114) / 1000
  return c >= 128 ? 'black' : 'white'
}

const Tags = ({ labels, setFilter }) => {
  return (
    <React.Fragment>
      {labels.map(label => (
        <Tag
          key={label.node_id}
          onClick={setFilter ? () => setFilter(label) : null}
          style={{
            backgroundColor: `#${label.color}`,
            color: `${chooseColor(label.color)}`,
            cursor: `${setFilter ? 'pointer' : null}`
          }}
        >
          {label.name}
        </Tag>
      ))}
    </React.Fragment>
  )
}

export default Tags
