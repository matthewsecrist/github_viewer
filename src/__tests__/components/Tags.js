import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import Tags from '../../components/Tags'

afterEach(cleanup)

const labels = [
  {
    name: 'mock-label-name',
    color: 'mock-label-color',
    node_id: 'mock-id'
  }
]

const mockSetFilter = jest.fn()

describe('<Tags />', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Tags labels={labels} setFilter={mockSetFilter} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('does not display when no labels are provided', () => {
    const { container } = render(<Tags labels={[]} setFilter={mockSetFilter} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('selects the filter when clicked', async () => {
    const { getByTestId } = render(
      <Tags labels={labels} setFilter={mockSetFilter} />
    )

    const el = await getByTestId('tags')
    fireEvent.click(el)

    expect(mockSetFilter).toHaveBeenCalled()
    expect(mockSetFilter).toHaveBeenCalledWith(
      expect.objectContaining({ name: 'mock-label-name' })
    )
  })
})
