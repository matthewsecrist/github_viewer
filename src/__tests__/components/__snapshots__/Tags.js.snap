// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> does not display when no labels are provided 1`] = `null`;

exports[`<Tags /> matches the snapshot 1`] = `
<span
  class="tag"
  data-testid="tags"
  style="color: white; cursor: pointer;"
>
  mock-label-name
</span>
`;
